qa/vagrant/build.gradle

```diff
@@ -17,8 +17,27 @@
  * under the License.
  */
 
-apply plugin: 'elasticsearch.vagrantsupport'
-apply plugin: 'elasticsearch.vagrant'
+plugins {
+  id 'groovy'
+  id 'application'
+  id 'com.github.johnrengelman.shadow' version '2.0.2'
+  id 'elasticsearch.build'
+  id 'elasticsearch.vagrantsupport'
+  id 'elasticsearch.vagrant'
+}
+
+dependencies {
+  compile localGroovy()
+  compile "org.hamcrest:hamcrest-all:${versions.hamcrest}"
+  runtime project(':libs:elasticsearch-core') // needs to be on the classpath for JarHell
+  packagingTest project(path: project.path, configuration: 'shadow')
+}
+
+shadowJar {
```

Comment:
I would stay away from shading if we can. What about having the executable jar contain a classpath entry, and having a simple "test" script be generated which invokes the jar?

Reply:
I'll give that a try. After thinking about it a little, dumping the test jars and all their dependencies into a directory and doing `java -cp /project/packaging/tests MainClass` is still really simple

Reply:
@rjernst I pushed https://github.com/elastic/elasticsearch/pull/29297/commits/c16706da663535be39254be3b7398cd0143fdbbe to have it copy the test jars and their deps separately (rather than an uberjar) and have the test projects specify a main class